File: game/player.py
import queue
import validation as validation
from core.vector2 import Vector2


class Player:

    def __init__(self, index):
        self.id = -1
        self.index = index
        self.opp_index = 1 if index == 0 else 0
        self.goal_row = 8 if index == 0 else 0
        self.opp_goal_row = 0 if index == 0 else 8
        self.starting_pos = Vector2(4, 0) if index == 0 else Vector2(4, 8)
        self.reset()

    def take_action(self, board_state):
        pass

    def get_valid_move_positions(self, board_state):
        return validation.get_valid_move_positions(self.pos, board_state)

    def reset(self):
        self.pos = self.starting_pos
        self.wall_count = 10

File: game/src/player.cc
#include "player.h"

Player::Player(const int id) : id_(id) {
}

void Player::SetIndex(const int index) {
    index_ = index;
    opp_index_ = (index == 0) ? 1 : 0;
    goal_row_ = (index == 0) ? 8 : 0;
    opp_goal_row_ = (index == 0) ? 0 : 8;
}

int Player::GetId() const {
    return id_;
}

int Player::GetGoalRow() const {
    return goal_row_;
}

std::vector<Vectori> Player::GetValidMoves(const BoardState& board_state) const {
    return board_state.GetValidMoves(index_);
}